### PR TITLE
Expose whole certificate chain to verify proc

### PR DESCRIPTION
**Pull request by gregnolle**
The existing Certificate structure only contains the certificate at the end of the chain. This PR exposes the full chain of issuers, and also adds full details of subject principal (beyond just the common name).

### CODE REVIEW

**Comment by kevinsawicki** [comment]
Would you mind updating the existing specs to assert these new fields?

File: atom/common/native_mate_converters/net_converter.cc [hunk 1]
```diff
@@ -45,8 +45,11 @@ v8::Local<v8::Value> Converter<scoped_refptr<net::X509Certificate>>::ToV8(
   std::string encoded_data;
   net::X509Certificate::GetPEMEncoded(
       val->os_cert_handle(), &encoded_data);
+
   dict.Set("data", encoded_data);
+  dict.Set("issuer", mate::ConvertToV8(isolate, val->issuer()));
```
Comment (deepak1556) [inline atom/common/native_mate_converters/net_converter.cc]: can be just `val->issuer()` , no need to explicitly call `mate::ConvertToV8`. Same for other places in this PR.

File: docs/api/structures/certificate-principal.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# CertificatePrincipal Object
```
Comment (MarshallOfSound) [inline docs/api/structures/certificate-principal.md]: /cc @zeke Needs to end up in `seeds.js`

Comment (zeke) [inline docs/api/structures/certificate-principal.md]: Ack. Thanks for pointing this out.

Comment (MarshallOfSound) [inline docs/api/structures/certificate-principal.md]: Keen for #6931 so we can stop caring about adding these every time 😄

**Comment by gregnolle** [comment]
I wasn't sure how far to go with the assertions since I see there a number of the existing fields don't have them in the specs. I've covered the same fields that are in the new structures though.

**Comment by zeke** [comment]
@gregnolle it looks like there are few linting errors:

```
  /home/travis/build/electron/electron/spec/api-app-spec.js:211:31: Extra semicolon.
  /home/travis/build/electron/electron/spec/api-app-spec.js:213:32: Extra semicolon.
  /home/travis/build/electron/electron/spec/api-session-spec.js:557:35: Extra semicolon.
  /home/travis/build/electron/electron/spec/api-session-spec.js:559:36: Extra semicolon.
```

**Comment by gregnolle** [comment]
I don't think the Travis build failure is due to anything on my branch (seems to be failing early on with an APT issue). Is there a way to trigger a rebuild?

**Comment by zeke** [comment]
restarted travis

**Comment by kevinsawicki** [comment]
@gregnolle thanks for this, rebased it and tweaked a few of the asserts.

The `issueCert` was `null` on the `select-client-certificate` event so I removed the assert for it to get the build passing. It was available on the `setCertificateVerifyProc` callback though.

Going to merge this in 👍 🚢

**Pull request merged**